rclpy/src/rclpy/service.cpp

```diff
@@ -174,24 +164,18 @@ service_take_request(py::capsule pyservice, py::object pyrequest_type)
   auto taken_request = std::unique_ptr<void, decltype(message_deleter)>(
     taken_request_ptr, message_deleter);
 
-  auto deleter = [](rmw_service_info_t * ptr) {PyMem_Free(ptr);};
-  auto header = std::unique_ptr<rmw_service_info_t, decltype(deleter)>(
-    static_cast<rmw_service_info_t *>(PyMem_Malloc(sizeof(rmw_service_info_t))),
-    deleter);
-  if (!header) {
-    throw std::bad_alloc();
-  }
+  rmw_service_info_t header;
```

Comment:
I opted for using a variable on the stack since this is a fairly small struct. Let me know if heap allocation would be preferred.